### PR TITLE
Use Parcel to build extension

### PR DESCRIPTION
**Pull request by buberdds**
Targeting side branch for extension work

- inject our CSP rules with extension related bits
- create separated task for unit tests (jest projects) and linters. We can run all of them or just a single project. Long term we can use this in CI where GH action/PR check could be triggered based on paths changes just to run necessary tests etc
- PR does not address e2e tests for extension. I think we will have to figure this out later. Cypress does not support other protocols than https/http, Playwright is limited to background page, here's sample for puppeteer with dynamic extension id https://github.com/oasisprotocol/oasis-wallet-web/commit/d4a784a9e73df9412ec415541e98fdddd2ba91a2, Metamask is using Selenium/Webdriver for example.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -195,7 +201,18 @@
...
     "web": {
-      "context": "browser"
+      "context": "browser",
+      "distDir": "./build"
+    },
+    "ext-dev": {
+      "distDir": "./build-ext",
+      "sourceMap": {
+        "inline": true,
+        "inlineSources": true
```
Comment (lukaw3d) [inline package.json:211]: This is recommended by https://parceljs.org/recipes/web-extension/#hmr.
I had this comment workaround in proof of concept: https://github.com/oasisprotocol/wallet-poc/blob/redux-ext-poc/package.json#L13-L19

File: package.json [hunk 2]
```diff
@@ -195,7 +201,18 @@
   },
   "targets": {
     "web": {
-      "context": "browser"
+      "context": "browser",
+      "distDir": "./build"
```
Comment (lukaw3d) [inline package.json:205]: ah, I missed this in https://github.com/oasisprotocol/oasis-wallet-web/pull/746: `build-without-minify.js` isn't used anymore and we didn't migrate.

We wanted better stack traces in bug reports: https://github.com/oasisprotocol/oasis-wallet-web/pull/498

File: package.json [hunk 3]
```diff
@@ -88,6 +97,7 @@
     "husky": "7.0.4",
     "i18next": "20.6.1",
     "i18next-browser-languagedetector": "6.1.2",
+    "parcel-transformer-env-variables-injection": "0.1.0",
```
Comment (lukaw3d) [inline package.json:100]: Nice

File: internals/getCsp.js [hunk 4]
```diff
@@ -3,8 +3,15 @@
 // - remove 'unsafe-inline' style by precomputing theme hash
 // - add report-uri to gather errors if anything was missed
 
+const cspEnhancement = `
+  frame-ancestors 
+    'self' 
+    https: http://localhost:* http://127.0.0.1:*;
+  `
```
Comment (lukaw3d) [inline internals/getCsp.js:10]: I wouldn't call it enhancement :D

**Review by lukaw3d — APPROVED** [review]

File: package.json (hunk 2 above)
Comment (buberdds) [reply package.json:205]: would like to keep optimisation for prod env, but if that's must have there is a `--no-optimize` flag. I will create PR to target master branch.

File: internals/getCsp.js (hunk 4 above)
Comment (buberdds) [reply internals/getCsp.js:10]: omg I forgot to push commit before merging. Anyway will change the variable name in a next extension PR.

**Pull request merged**